pkg/helmreconciler/patching.go

```diff
@@ -0,0 +1,158 @@
+package helmreconciler
+
+import (
+	"context"
+	"fmt"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/jsonmergepatch"
+	"k8s.io/apimachinery/pkg/util/mergepatch"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/kubernetes/pkg/kubectl"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreatePatch creates a patch based on the current and updated versions of an object
+func (h *HelmReconciler) CreatePatch(current, updated runtime.Object) (Patch, error) {
+	var retVal Patch
+	patch := &basicPatch{client: h.client}
+	currentAccessor, err := meta.Accessor(current)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("cannot create object accessor for current object:\n%v", current))
```

Comment:
any reason not to just use return nil, fmt.Errorf(), here and elsewhere?

Reply:
I wanted to collect the upstream error, which is often useful for debugging and support.